Src/AutoFixture/DefaultRelays.cs

```diff
@@ -28,6 +28,7 @@ public virtual IEnumerator<ISpecimenBuilder> GetEnumerator()
             yield return new FieldRequestRelay();
             yield return new FiniteSequenceRelay();
             yield return new SeedIgnoringRelay();
+            yield return new RangedRequestToRangedNumberRequestRelay();
```

Comment:
The complexity will start showing up once we run into more cases, other than turning a ranged request to a ranged number request. Then, I could imagine a bloated `DefaultRelays`

```c#
...
yield return new FiniteSequenceRelay();
yield return new SeedIgnoringRelay();

yield return new RangedRequestToRangedNumberRequestRelay();
yield return new RangedRequestToXyzRequestRelay();
yield return new RangedRequestToAbcRequestRelay();

yield return new MethodInvoker(
                  new CompositeMethodQuery(
...
```

Instead, I could imagine this relay of relays being a [Composite](https://en.wikipedia.org/wiki/Composite_pattern):

```c#
yield return new RangedRequestRelay(
	new NumericRangedRequestRelay(),
	new XyzRangedRequestRelay(),
	new AbcRangedRequestRelay());
```

Perhaps, internally it'll use the [Specification](https://en.wikipedia.org/wiki/Specification_pattern) like we do in other places of this code base.

Reply:
Thanks for the review.

Well, I understand your concern and indeed after we add a set of builders in a row, they will "pollute" the chain a bit. However, I don't see too much options we have. 

Currently we have a situation when `RangedRequest` is relayed to the `RangedNumberRequest`. It might look that `RangedRequest` will be always relayed to the more specific requests only, but that is not necessarily so. For instance, I'm going to fix [the issue with enums](722) in a way, so we have the following path: `RangedRequest -> EnumRangedRequestRelay -> RangedNumberRequest`. In this path one "sibling" relays to other siblings. So all the builders should be able to intercept the request in case of recursive calls. 

Also we would like to allow users to add their own customizations to handle the `RangedRequest` of custom types, therefore `RangedRequestRelay` should perform a recursive request, instead of delegating to its children only.

Unfortunately, while I'm not sure I see how the suggested design allows to achieve all that stuff in a graceful way. Could you please describe that in more detail if you have in idea in mind?

Also we cannot simply wrap all the range related builders to the `CompositeBuilder` (to logically group them) because [we have a customization](https://github.com/AutoFixture/AutoFixture/blob/0babb907b0fd7524201e99d1d5e9189e22c24599/Src/AutoFixture/NoDataAnnotationsCustomization.cs#L30) which removes the `RangedRequestRelay` from the fixture (for the performance sake).

Would be really great if you see a better way, but currently it looks like that current way is the  most appropriate one..